# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1261362282 28800
#      Sun Dec 20 18:24:42 2009 -0800
# Node ID 411325ce32d7c232de33dda6f211397d32518978
# Parent  c5e9931c7ba7ff3e336b3b5581d9f7e5d61462ef
Better warnings when gnuplot is not installed but native graphics are available

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-20  Rik <octave@nomad.inbox5.com>
+
+	* configure.ac, acinclude.m4: Better warnings when gnuplot is not
+	installed but native graphics are available
+
 2009-12-19  Rik <octave@nomad.inbox5.com>
 
 	* emacs/module.mk, examples/Makefile.am, info-emacs-info,
 	info-emacs-octave-help: Distribute emacs specific files from emacs
 	directory rather than examples directory
 
 2009-12-19  Rik <octave@nomad.inbox5.com>
 
diff --git a/acinclude.m4 b/acinclude.m4
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -630,32 +630,19 @@ if test "$cross_compiling" = yes; then
   AC_MSG_RESULT(assuming $GNUPLOT exists on $canonical_host_type host)
 else
   AC_CHECK_PROGS(GNUPLOT, $gp_names)
   if test -z "$GNUPLOT"; then
     warn_gnuplot=yes
 
     GNUPLOT="$gp_default"
 
-    ## If you change this text, be sure to also copy it to the set of
-    ## warnings at the end of the script
-
-    AC_MSG_WARN([I didn't find gnuplot.  It isn't necessary to have gnuplot])
-    AC_MSG_WARN([installed, but you won't be able to use any of Octave's])
-    AC_MSG_WARN([plotting commands without it.])
-    AC_MSG_WARN([])
-    AC_MSG_WARN([If gnuplot is installed but it isn't in your path, you can])
-    AC_MSG_WARN([tell Octave where to find it using the gnuplot_binary])
-    AC_MSG_WARN([function.  For example,])
-    AC_MSG_WARN([])
-    AC_MSG_WARN([gnuplot_binary ("/full/name/of/gnuplot/binary")])
-    AC_MSG_WARN([])
-    AC_MSG_WARN([at the Octave prompt.])
-    AC_MSG_WARN([])
-    AC_MSG_WARN([Setting default value to $GNUPLOT])
+    AC_MSG_WARN([gnuplot not found.  It isn't necessary to have gnuplot])
+    AC_MSG_WARN([installed, but without native graphics or gnuplot])
+	 AC_MSG_WARN([you won't be able to use any of Octave's plotting commands.])
   fi
 fi
 AC_SUBST(GNUPLOT)
 ])
 dnl
 dnl Is gperf installed?
 dnl
 dnl OCTAVE_PROG_GPERF
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2385,39 +2385,16 @@ if test -n "$warn_texi2pdf"; then
   warn_msg_printed=true
 fi
 
 if test -n "$warn_64_bit"; then
   AC_MSG_WARN([$warn_64_bit])
   warn_msg_printed=true
 fi
 
-if test -n "$warn_gnuplot"; then
-
-  ## If you change this text, be sure to also change the corresponding
-  ## set of warnings above.
-
-  AC_MSG_WARN([])
-  AC_MSG_WARN([I didn't find gnuplot.  It isn't necessary to have gnuplot])
-  AC_MSG_WARN([installed, but you won't be able to use any of Octave's])
-  AC_MSG_WARN([plotting commands without it.])
-  AC_MSG_WARN([])
-  AC_MSG_WARN([If gnuplot is installed but it isn't in your path, you can])
-  AC_MSG_WARN([tell Octave where to find it using the gnuplot_binary])
-  AC_MSG_WARN([function.  For example,])
-  AC_MSG_WARN([])
-  AC_MSG_WARN([gnuplot_binary ("/full/name/of/gnuplot/binary")])
-  AC_MSG_WARN([])
-  AC_MSG_WARN([at the Octave prompt.])
-  AC_MSG_WARN([])
-  AC_MSG_WARN([Setting default value to $GNUPLOT])
-  AC_MSG_WARN([])
-  warn_msg_printed=true
-fi
-
 if $USE_64_BIT_IDX_T; then
   AC_MSG_WARN([])
   AC_MSG_WARN([You used the EXPERIMENTAL --enable-64 option.])
   AC_MSG_WARN([Are you sure that is what you want to do?])
   AC_MSG_WARN([])
   AC_MSG_WARN([You must ensure that the Fortran compiler generates])
   AC_MSG_WARN([code with 8 byte signed INTEGER values, and that your])
   AC_MSG_WARN([BLAS and LAPACK libraries are compiled to use 8 byte])
@@ -2451,22 +2428,51 @@ if test -n "$warn_fltk_opengl"; then
   warn_msg_printed=true
 fi
 
 if $native_graphics; then
   true;
 else
   AC_MSG_WARN([])
   AC_MSG_WARN([I didn't find the necessary libraries to compile native])
-  AC_MSG_WARN([graphics. It isn't necessary to have native graphics])
-  AC_MSG_WARN([but you will have to use gnuplot or you won't be able])
-  AC_MSG_WARN([to use any of Octave's plotting commands])
+  AC_MSG_WARN([graphics.  It isn't necessary to have native graphics,])
+  AC_MSG_WARN([but you will need to have gnuplot installed or you won't])
+  AC_MSG_WARN([be able to use any of Octave's plotting commands])
   AC_MSG_WARN([])
   warn_msg_printed=true
 fi
 
+if test -n "$warn_gnuplot"; then
+  if $native_graphics; then
+    AC_MSG_WARN([])
+    AC_MSG_WARN([I didn't find gnuplot.  Plotting commands will use the])
+    AC_MSG_WARN([native graphics backend.])
+  else
+    AC_MSG_WARN([I didn't find gnuplot.  It isn't necessary to have gnuplot])
+    AC_MSG_WARN([installed, but you won't be able to use any of Octave's])
+    AC_MSG_WARN([plotting commands without it.])
+  fi
+  AC_MSG_WARN([])
+  AC_MSG_WARN([If gnuplot is installed, but isn't in your path, you can])
+  AC_MSG_WARN([tell Octave where to find it by using the gnuplot_binary])
+  AC_MSG_WARN([function.  For example,])
+  AC_MSG_WARN([])
+  AC_MSG_WARN([gnuplot_binary ("/full/path/and/name/of/gnuplot/binary")])
+  AC_MSG_WARN([])
+  AC_MSG_WARN([at the Octave prompt.])
+  AC_MSG_WARN([])
+  AC_MSG_WARN([Setting default value to $GNUPLOT])
+  AC_MSG_WARN([])
+
+  warn_msg_printed=true
+fi
+
+
+
+
+
 if $warn_msg_printed; then
   AC_MSG_NOTICE([])
   AC_MSG_NOTICE([NOTE: libraries may be skipped if a library is not found OR])
   AC_MSG_NOTICE([NOTE: if the library on your system is missing required features.])
 fi
 
 ### End of configure.
